resqpy/olio/relperm.py

```diff
@@ -0,0 +1,316 @@
+         relp_opp_col = [x for x in df.columns if (x[0:2] == 'Kr') & (x[-1] != sat_col[-1])][0]
+         assert (df[sat_col].is_monotonic and df[relp_corr_col].is_monotonic and df[
+             relp_opp_col].is_monotonic_decreasing) or \
+             (df[sat_col].is_monotonic_decreasing and df[relp_corr_col].is_monotonic_decreasing and
+                 df[relp_opp_col].is_monotonic), f'{sat_col, relp_corr_col, relp_opp_col} combo is not monotonic'
+         if 'Pc' in df.columns:
+            assert df['Pc'].dropna().is_monotonic or df['Pc'].dropna(
+            ).is_monotonic_decreasing, 'Pc values are not monotonic'
+         for col in ['Sw', 'Sg', 'So', 'Krw', 'Krg', 'Kro']:
+            if col in df.columns:
+               assert df[col].min() >= 0 and df[col].max() <= 1, f'{col} is not within the range 0-1'
+
+      super().__init__(model,
+                       uuid = uuid,
+                       support_root = None,
+                       df = df,
+                       uom_list = uom_list,
+                       realization = realization,
+                       title = title,
+                       column_lookup_uuid = column_lookup_uuid,
+                       uom_lookup_uuid = uom_lookup_uuid)
+      self.phase_combo = phase_combo
+      self.low_sal = low_sal
+      self.table_index = table_index
+
+   def interpolate_point(self, saturation, kr_or_pc_col):
+      """Returns a tuple of the saturation value and the corresponding
+        interpolated relative permeability or capillary pressure value.
+
+        arguments:
+            saturation (float): the saturation at which the relative permeability or cap. pressure will be interpolated
+            kr_or_pc_col (str): the column name of the parameter to be interpolated
+        returns:
+            tuple of float, the first element is the saturation and the second element is the interpolated value
+           """
+      df = self.df.copy()
+      assert kr_or_pc_col.capitalize(
+      ) in df.columns and kr_or_pc_col.capitalize() != df.columns[0], 'incorrect column name provided for interpolation'
+      if kr_or_pc_col == 'PC':
+         df = df[df['PC'].notnull()]
+      sat_col = df.columns[0]
+      x = df[sat_col]
+      y = df[kr_or_pc_col]
+      f = interpolate.interp1d(x, y, kind = 'linear', assume_sorted = True)
+      x_new = saturation
+      assert x.min() <= x_new <= x.max(), 'saturation value is outside the interpolation range'
+      y_new = f(x_new)
+      return saturation, y_new.item()
+
+   def df_to_text(self, filepath, filename):
+      """Creates text file from a dataframe of relative permeability and capillary pressure data.
+
+        arguments:
+            filepath (str): location where new text file is written to
+            filename (str): name of the new text file
+        returns:
+            tuple of float, the first element is the saturation and the second element is the interpolated value
+           """
```

Comment:
I suggest a note saying that the format is for Nexus (though other formats could be supported in future).
If the word Nexus does appear anywhere, please add a trademark comment at the top of the module, something like:
# Nexus is a registered trademark of Halliburton

Reply:
Done! I've added the trademark comment to the top of the relperm.py file